src/main/java/com/streamr/broker/Main.java

```diff
@@ -4,20 +4,29 @@
 import com.streamr.broker.kafka.KafkaListener;
 import com.streamr.broker.redis.RedisReporter;
 import com.streamr.broker.stats.LoggedStats;
+import com.streamr.broker.stats.MetricsStats;
+import com.streamr.broker.stats.Stats;
+
+import java.util.ArrayList;
 
 public class Main {
 
-	private static LoggedStats stats;
+	private static LoggedStats loggedStats;
 
 	public static void main(String[] args) {
 		BrokerProcess brokerProcess = new BrokerProcess(Config.QUEUE_SIZE);
-		if (Config.METRICS_STREAM_ID.equals("")) {
-			stats = new LoggedStats();
-		} else {
-			stats = new LoggedStats(Config.METRICS_INTERVAL_IN_SECS, Config.METRICS_STREAM_ID,
-					Config.METRICS_API_KEY, Config.METRICS_WS_URL,Config.METRICS_REST_URL);
+		ArrayList<Stats> stats = new ArrayList<>();
+		if (!Config.METRICS_STREAM_ID.equals("")) {
+			stats.add(new MetricsStats(Config.METRICS_INTERVAL_IN_SECS, Config.METRICS_STREAM_ID,
+					Config.METRICS_API_KEY, Config.METRICS_WS_URL,Config.METRICS_REST_URL));
+		}
+		loggedStats = new LoggedStats(Config.STATS_INTERVAL_IN_SECS);
+		stats.add(loggedStats);
+		Stats[] statsArray = new Stats[stats.size()];
+		for (int i = 0; i < stats.size(); i++) {
+			statsArray[i] = stats.get(i);
 		}
-		brokerProcess.setStats(stats, Config.STATS_INTERVAL_IN_SECS, Config.METRICS_INTERVAL_IN_SECS);
+		brokerProcess.setStats(statsArray);
```

Comment:
Cleanup suggestion: Can use this and remove lines 25-28

```suggestion
		brokerProcess.setStats(stats.toArray());
```